Internal static library leaks to the ldflags when using gnome.gtkdoc()

Similar to #3934: additional uninstalled static libraries end up on the linker flags for shared libraries, causing build failures when running `gtkdoc-scangobj`:

```
ERROR: 'gtkdoc-scangobj' failed with status 1
/usr/bin/ld: cannot find -lgeoclue-public-api
collect2: error: ld returned 1 exit status
2018-07-24 18:53:45,120:scangobj.py:execute_command:1196:WARNING:Linking scanner failed: 1, command: ccache cc libgeoclue-scan.o -L/home/ebassi/gnome/source/geoclue/_build/libgeoclue -L/home/ebassi/gnome/install/lib -Wl,-rpath,/home/ebassi/gnome/install/lib -Wl,-rpath,/home/ebassi/gnome/source/geoclue/_build/libgeoclue -lgeoclue-2 -lgeoclue-public-api -lglib-2.0 -lgio-2.0 -lgobject-2.0 -o libgeoclue-scan
```

`geoclue-public-api` is a static library defined as an internal dependency of `geoclue-2`.

I have the same issue with the [wip evince's meson port](https://gitlab.gnome.org/GNOME/evince/merge_requests/80).

`evince` binary dependencies include [4 static libraries](https://gitlab.gnome.org/GNOME/evince/merge_requests/80/diffs?commit_id=be173beae37452ee989eb928c9307bc1d3594c1c#949ff4ad2adba0d5a2c089cb7ad6a430eabf853c_0_67): [libevmisc](https://gitlab.gnome.org/GNOME/evince/merge_requests/80/diffs?commit_id=be173beae37452ee989eb928c9307bc1d3594c1c#b803f5b061c0abbb2f39adafc658e2795982ad43_0_7), [libevproperties](https://gitlab.gnome.org/GNOME/evince/merge_requests/80/diffs?commit_id=be173beae37452ee989eb928c9307bc1d3594c1c#cbba70be4c8e1bbb014c7d4d256910678a4300c6_0_1), [libgd](https://gitlab.gnome.org/GNOME/evince/merge_requests/80/diffs?commit_id=be173beae37452ee989eb928c9307bc1d3594c1c#6680ab60a248fd798053abc72673eb8a89283593_0_11) and [libgimpcellrenderertoggle](https://gitlab.gnome.org/GNOME/evince/merge_requests/80/diffs?commit_id=be173beae37452ee989eb928c9307bc1d3594c1c#cdf62a975807839fb7938675aa6c9ed871d8c0a1_0_1).

These are also used when generating [gtkdoc documentation](https://gitlab.gnome.org/GNOME/evince/merge_requests/80/diffs?commit_id=be173beae37452ee989eb928c9307bc1d3594c1c#c5e3299722c471b70f0186425f2cc673f98f00f4_0_30), that produces the following error:

```
ERROR: 'gtkdoc-scangobj' failed with status 1
/usr/bin/ld: cannot find -levmisc
/usr/bin/ld: cannot find -levproperties
/usr/bin/ld: cannot find -lgd
/usr/bin/ld: cannot find -lgimpcellrenderertoggle
collect2: error: ld returned 1 exit status
```

--- a/mesonbuild/modules/gnome.py
+++ b/mesonbuild/modules/gnome.py
@@ -331,14 +331,15 @@ class GnomeModule(ExtensionModule):
                 for lib in dep.libraries:
                     if hasattr(lib, 'held_object'):
                         lib = lib.held_object
-                    internal_ldflags.update(self._get_link_args(state, lib, depends, include_rpath))
-                    libdepflags = self._get_dependencies_flags(lib.get_external_deps(), state, depends, include_rpath,
-                                                               use_gir_args, True)
-                    cflags.update(libdepflags[0])
-                    internal_ldflags.update(libdepflags[1])
-                    external_ldflags.update(libdepflags[2])
-                    external_ldflags_nodedup += libdepflags[3]
-                    gi_includes.update(libdepflags[4])
+                    if isinstance(lib, build.SharedLibrary):
+                        internal_ldflags.update(self._get_link_args(state, lib, depends, include_rpath))
+                        libdepflags = self._get_dependencies_flags(lib.get_external_deps(), state, depends, include_rpath,
+                                                                   use_gir_args, True)
+                        cflags.update(libdepflags[0])
+                        internal_ldflags.update(libdepflags[1])
+                        external_ldflags.update(libdepflags[2])
+                        external_ldflags_nodedup += libdepflags[3]
+                        gi_includes.update(libdepflags[4])
                 extdepflags = self._get_dependencies_flags(dep.ext_deps, state, depends, include_rpath,
                                                            use_gir_args, True)
                 cflags.update(extdepflags[0])

